src/DssDirectDepositAaveDai.sol

```diff
@@ -255,9 +260,14 @@ contract DssDirectDepositAaveDai {
             return;
         }
 
+
         require(int256(amount) >= 0, "DssDirectDepositAaveDai/overflow");
 
         uint256 scaledPrev = adai.scaledBalanceOf(address(this));
+        uint256 interestIndex = pool.getReserveNormalizedIncome(address(dai));
+        uint256 scaledNew = _add(scaledPrev, _rdiv(amount, interestIndex));
+
+        require(_rdiv(scaledNew, availableLiquidity) <= pie, "DssDirectDepositAaveDai/share-too-large");
```

Comment:
This should probably compare the unscaled version to `availableLiquidity` as the scaled version is not the same as the amount of actual Dai available to the D3M. Also this needs to deal with the case where `availableLiquidity == 0` as this will be a division by zero which will fail, but maybe it's better to handle that explicitly.